packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/query/QueryPredicateBuilder.kt

```diff
@@ -25,17 +26,27 @@ import com.amplifyframework.core.model.query.predicate.QueryPredicateOperation.n
 
 class QueryPredicateBuilder {
     companion object {
+        /**
+         * Use only within [QueryOptionsBuilder]
+         */
         @JvmStatic
-        fun fromSerializedMap(serializedMap: Map<String, Any>?): QueryPredicate? {
+        fun fromSerializedMap(serializedMap: Map<String, Any>?, modelSchema: ModelSchema?): QueryPredicate? {
             if (serializedMap == null) {
                 return null
             }
 
             if (serializedMap.containsKey("queryPredicateOperation")) {
-                val queryPredicateOperationMap: Map<String, Any> = serializedMap["queryPredicateOperation"].safeCastToMap()!!
-                val field = queryPredicateOperationMap["field"] as String
+                val queryPredicateOperationMap: Map<String, Any> =
+                    serializedMap["queryPredicateOperation"].safeCastToMap()!!
+                var field = queryPredicateOperationMap["field"] as String
+
+                if (modelSchema?.associations?.containsKey(field) == true) {
```

Comment:
Should this also check that `association.associatedType` is belongsTo?

Reply:
Good catch, it should work only with `belongsTo`.

Though I found another issue within the generated model while testing `hasOne`. Use below schema as example:
```graphql
type Post @model{
  id: ID!
  title: String!
  content: String!
  reviewerId: ID
  reviewer: Person @connection(fields: [reviewerId])
}

type Person {
  id: ID!
  name: String!
}
```

Generated `Post` model has correct fields

```
class Post extends Model {
  ...
  final String? _reviewerId;
  final Person? _reviewer;
```

However, the generated model marks `reviewer` as a `QueryField`. 

```
  static final QueryField REVIEWERID = QueryField(fieldName: "reviewerId");
  static final QueryField REVIEWER = QueryField(
    fieldName: "reviewer",
    fieldType: ModelFieldType(ModelFieldTypeEnum.model, ofModelName: (Person).toString()));
```
This is incorrect, as with `hasOne` `reviewerId` should be the query field.

I will update the PR tomorrow, and open an issue for codegen to fix this newly found issue.

Reply:
Correction: it should check `association.name` to determine relationship.

In addition, if the codegen issue gets fixed, theoretically the check of `association.name == "belongsTo"` could be omitted. Will add the guard regardless.

Reply:
Good call - I think the codegen issue could be similar to what Jordan noticed here: https://github.com/aws-amplify/amplify-ios/issues/1107#issuecomment-868538507. May want to consult with him, as this could be the expected codegen.

Reply:
Thanks. That should be a different issue generating `hasOne` vs `belongsTo`. This one should be isolated as not generating `QueryField` when it's `hasOne`, I think.